app/routes.py

```diff
@@ -1 +1,269 @@
+from app.models.goal import Goal
+from app import db
+import datetime
+import os
+import requests
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+path = "https://slack.com/api/chat.postMessage"
+
+#TASK ROUTES
+
+def validate_task_id(task_id):
+    try:
+        task_id =  int(task_id)
+    except:
+        return abort(make_response(jsonify({'message': f"Invalid task: {task_id}"}), 400))
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return abort(make_response(jsonify({'message': f"Could not find task with id {task_id}"}), 404))
+    return task       
+
+@tasks_bp.route('', methods=['POST'])
+def create_one_task():
+    request_body = request.get_json()
+    if 'title' not in request_body or 'description' not in request_body:
+        return {"details": "Invalid data"}, 400
+
+    if 'completed_at' in request_body:
+        new_task = Task(title=request_body["title"],
+                    description=request_body["description"],
+                    completed_at=request_body['completed_at'])
+    else:
+        new_task = Task(title=request_body["title"],
+                    description=request_body["description"])
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {
+        "task": {
+        "id": new_task.task_id,
+        "title": new_task.title,
+        "description": new_task.description,
+        "is_complete": isinstance(new_task.completed_at, datetime.datetime)
+    }}, 201
+
+@tasks_bp.route('', methods=['GET'])
+def get_all_tasks():
+    params = request.args
+    task_response = []
+
+    if "sort" in params:
+        sort_order = request.args.get('sort')
+        if sort_order == "asc":
+            tasks = Task.query.order_by(Task.title.asc())
+        elif sort_order =="desc":
+            tasks = Task.query.order_by(Task.title.desc())
+    
+    else:
+        tasks = Task.query.all()
+    
+    for task in tasks:
+        task_response.append({
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": isinstance(task.completed_at, datetime.datetime)
+    })
+
+    return jsonify(task_response), 200
+
+@tasks_bp.route('/<task_id>', methods=['GET'])
+def get_one_task(task_id):
+    one_task = validate_task_id(task_id)
+    print(one_task.goal_id)
+    if one_task.goal_id:
+        response = {"goal_id": one_task.goal_id,
+                    "id": one_task.task_id,
+                    "title": one_task.title,
+                    "description": one_task.description,
+                    "is_complete": isinstance(one_task.completed_at, datetime.datetime)}
+    else:
+        response = {
+                "id": one_task.task_id,
+                "title": one_task.title,
+                "description": one_task.description,
+                "is_complete": isinstance(one_task.completed_at, datetime.datetime)
+            }
```

Comment:
Stylistically I recommend rewriting this section into something like:
```suggestion
    response = {"id": one_task.task_id,
                    "title": one_task.title,
                    "description": one_task.description,
                    "is_complete": isinstance(one_task.completed_at, datetime.datetime)}
    if one_task.goal_id:
        response["goal_id"] = one_task.goal_id
```
The structure of the dictionary is the same in both if branches, the only difference is that goal_id is included if it's not None.  As written, it requires a much closer reading of the code to determine this.  In addition, pulling the sections that are the same out of the if block makes it possible to DRY the code.